File: Java/src/main/java/connectn/GameStatus.java
package connectn;

/**
 * Represents the current status of a Connect-N game.
 * 
 * <p>The game can be in progress, won by either player, or ended in a draw.
 * 
 * @author CS5004 Teaching Team
 * @version 1.0
 */
public enum GameStatus {
    /**
     * The game is still in progress and moves can be made.
     */
    IN_PROGRESS,
    
    /**
     * Player 1 has won the game by connecting N tokens.
     */
    PLAYER_ONE_WON,
    
    /**
     * Player 2 has won the game by connecting N tokens.
     */
    PLAYER_TWO_WON,
    
    /**
     * The game has ended in a draw (board is full with no winner).
     */
    DRAW
}

File: Python/connectn/game_status.py
"""Game status enumeration for Connect-N game."""

from enum import Enum, auto


class GameStatus(Enum):
    """
    Represents the current status of a Connect-N game.
    
    Attributes:
        IN_PROGRESS: Game is still being played
        PLAYER_ONE_WON: Player 1 has won the game
        PLAYER_TWO_WON: Player 2 has won the game
        DRAW: Game ended in a draw (board full, no winner)
    """
    
    IN_PROGRESS = auto()
    PLAYER_ONE_WON = auto()
    PLAYER_TWO_WON = auto()
    DRAW = auto()
    
    def __str__(self) -> str:
        """String representation of the game status."""
        return self.name.replace('_', ' ').title()
